Ginger/GingerCoreNET/Database/NoSqlBase/GingerCassandra.cs

```diff
@@ -44,25 +46,62 @@ public override bool Connect()
             try
             {
                 string queryTimeoutString = "querytimeout=";
+                string SLL = "SSL=";
+                string sslValue = null;
                 int queryTimeout = 20000;//default timeout (20 seconds).
-                if (Db.DatabaseOperations.TNSCalculated.ToLower().Contains(queryTimeoutString.ToLower()))
+                string[] queryArray = Db.DatabaseOperations.TNSCalculated.Split(';');
+                if (queryArray[1].ToLower().Contains(queryTimeoutString.ToLower()))
                 {
-                    string queryTimeoutValue = Db.DatabaseOperations.TNSCalculated.Substring(Db.DatabaseOperations.TNSCalculated.ToLower().IndexOf(queryTimeoutString.ToLower()) + queryTimeoutString.Length);
+                    string queryTimeoutValue = queryArray[1].Substring(queryArray[1].ToLower().IndexOf(queryTimeoutString.ToLower()) + queryTimeoutString.Length);
                     queryTimeout = Convert.ToInt32(queryTimeoutValue) * 1000;
                 }
 
-                string[] HostKeySpace = Db.DatabaseOperations.TNSCalculated.Split('/');
+                if (queryArray[2].ToLower().Contains(SLL.ToLower()))
+                {
+                    sslValue = queryArray[2].Substring(queryArray[2].ToLower().IndexOf(SLL.ToLower()) + SLL.Length);
+                }
+                string[] HostKeySpace = queryArray[0].Split('/');
```

Comment:
as we discussed on line no 63, the connection string may also contain http://<hostName>:<portNo>/<KeySpaceName>
so make it string[] HostKeySpace = queryArray[0].ToLower().Replace("http://","").Split('/');